src/include/rbd/librbd.h

```diff
@@ -569,6 +569,12 @@ CEPH_RBD_API ssize_t rbd_write(rbd_image_t image, uint64_t ofs, size_t len,
  */
 CEPH_RBD_API ssize_t rbd_write2(rbd_image_t image, uint64_t ofs, size_t len,
                                 const char *buf, int op_flags);
+
+CEPH_RBD_API ssize_t rbd_writesame(rbd_image_t image, uint64_t off, size_t write_len,
+                              const char *buf, size_t data_len);
+CEPH_RBD_API ssize_t rbd_writesame2(rbd_image_t image, uint64_t off, size_t write_len,
```

Comment:
No need to add ```writesame2``` API methods --- just add the ```op_flags``` to the new call.  The xyz2 methods are added to ensure the API doesn't break.